src/torrent.cpp

```diff
@@ -5291,12 +5291,20 @@ namespace libtorrent
 		TORRENT_ASSERT(std::count(m_peers_to_disconnect.begin()
 			, m_peers_to_disconnect.end(), p) == 0);
 
-		std::weak_ptr<torrent> weak_t = shared_from_this();
-		m_peers_to_disconnect.push_back(p);
-		m_deferred_disconnect.post(m_ses.get_io_service(), [=]() {
-			std::shared_ptr<torrent> t = weak_t.lock();
-			if (t) t->on_remove_peers();
-		});
+		// we only schedule the peer for actual removal if in fact
+		// it is attached to this torrent, since this method can
```

Comment:
sorry for nit-picking. would you mind changing "method" to "function"? (or "member function")